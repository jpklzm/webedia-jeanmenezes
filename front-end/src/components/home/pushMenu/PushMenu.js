import React, { Component } from 'react';
import { push as Menu } from 'react-burger-menu';
import { Link } from 'react-router-dom';
import styles from './styles.scss';

// Mobile Menu

// Necessary style json for third-party component
const menuStyles = {
  bmBurgerButton: {
    position: 'absolute',
    width: '60px',
    height: '60px',
    left: '36px',
    top: '36px',
  },
  bmCrossButton: {
    height: '63px',
    width: '63px',
  },
  bmCross: {
    background: '#fff',
  },
  bmMenu: {
    background: '#fff',
    padding: '2.5em 1.5em 0',
    fontSize: '1.15em',
  },
  bmMorphShape: {
    fill: '#f0f0f0',
  },
  bmItemList: {
    color: '#fe783b',
    padding: '0.8em',
  },
  bmOverlay: {
    // background: 'rgba(0, 0, 0, 0.3)',
  },
};

class PushMenu extends Component {
  render() {
    return (
      <Menu
        styles={menuStyles}
        pageWrapId={'page-wrap'}
        customBurgerIcon={<img src="/src/assets/menu.svg" />}
        customCrossIcon={<img src="/src/assets/back.svg" />}
      >
        <Link to="/" className={styles.menuText}>Home</Link>
        <a href="#" className={styles.menuText}>Lorem Ipsum</a>
        <a href="#" className={styles.menuText}>Lorem Ipsum</a>
        <a href="#" className={styles.menuText}>Lorem Ipsum</a>
        <a href="#" className={styles.menuText}>Lorem Ipsum</a>
        <a href="#" className={styles.menuText}>Lorem Ipsum</a>
        <a href="#" className={styles.menuText}>Lorem Ipsum</a>
        <a href="#" className={styles.menuText}>Lorem Ipsum</a>
        <a href="#" className={styles.menuText}>Lorem Ipsum</a>
        <Link to="/admin" className={styles.menuText}>Admin</Link>
      </Menu>
    );
  }
}

export default PushMenu;
