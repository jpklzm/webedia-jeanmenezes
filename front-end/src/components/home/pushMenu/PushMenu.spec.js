import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import PushMenu from './PushMenu';

test('Snapshot', () => {
  const component = renderer.create(
    <MemoryRouter>
      <PushMenu />
    </MemoryRouter>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
