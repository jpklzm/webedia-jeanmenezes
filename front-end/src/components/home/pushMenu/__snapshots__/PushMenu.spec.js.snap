// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot 1`] = `
<div>
  <div
    className="bm-overlay "
    onClick={[Function]}
    style={
      Object {
        "MozTransform": "translate3d(100%, 0, 0)",
        "MsTransform": "translate3d(100%, 0, 0)",
        "OTransform": "translate3d(100%, 0, 0)",
        "WebkitTransform": "translate3d(100%, 0, 0)",
        "background": "rgba(0, 0, 0, 0.3)",
        "height": "100%",
        "opacity": 0,
        "position": "fixed",
        "transform": "translate3d(100%, 0, 0)",
        "transition": "opacity 0.3s, transform 0s 0.3s",
        "width": "100%",
        "zIndex": 1,
      }
    }
  />
  <div
    className="bm-menu-wrap "
    id=""
    style={
      Object {
        "MozTransform": "translate3d(-100%, 0, 0)",
        "MsTransform": "translate3d(-100%, 0, 0)",
        "OTransform": "translate3d(-100%, 0, 0)",
        "WebkitTransform": "translate3d(-100%, 0, 0)",
        "height": "100%",
        "position": "fixed",
        "right": "inherit",
        "transform": "translate3d(-100%, 0, 0)",
        "transition": "all 0.5s",
        "width": "300px",
        "zIndex": 2,
      }
    }
  >
    <div
      className="bm-menu "
      style={
        Object {
          "background": "#f0f0f0",
          "boxSizing": "border-box",
          "fontSize": "1.15em",
          "height": "100%",
          "overflow": "auto",
          "padding": "2.5em 1.5em 0",
        }
      }
    >
      <nav
        className="bm-item-list "
        style={
          Object {
            "color": "#fe783b",
            "height": "100%",
            "padding": "0.8em",
          }
        }
      >
        <a
          className="menuText"
          href="/"
          onClick={[Function]}
          style={
            Object {
              "display": "block",
              "outline": "none",
            }
          }
        >
          Home
        </a>
        <a
          className="menuText"
          href="#"
          style={
            Object {
              "display": "block",
              "outline": "none",
            }
          }
        >
          Lorem Ipsum
        </a>
        <a
          className="menuText"
          href="#"
          style={
            Object {
              "display": "block",
              "outline": "none",
            }
          }
        >
          Lorem Ipsum
        </a>
        <a
          className="menuText"
          href="#"
          style={
            Object {
              "display": "block",
              "outline": "none",
            }
          }
        >
          Lorem Ipsum
        </a>
        <a
          className="menuText"
          href="#"
          style={
            Object {
              "display": "block",
              "outline": "none",
            }
          }
        >
          Lorem Ipsum
        </a>
        <a
          className="menuText"
          href="#"
          style={
            Object {
              "display": "block",
              "outline": "none",
            }
          }
        >
          Lorem Ipsum
        </a>
        <a
          className="menuText"
          href="#"
          style={
            Object {
              "display": "block",
              "outline": "none",
            }
          }
        >
          Lorem Ipsum
        </a>
        <a
          className="menuText"
          href="#"
          style={
            Object {
              "display": "block",
              "outline": "none",
            }
          }
        >
          Lorem Ipsum
        </a>
        <a
          className="menuText"
          href="#"
          style={
            Object {
              "display": "block",
              "outline": "none",
            }
          }
        >
          Lorem Ipsum
        </a>
        <a
          className="menuText"
          href="/admin"
          onClick={[Function]}
          style={
            Object {
              "display": "block",
              "outline": "none",
            }
          }
        >
          Admin
        </a>
      </nav>
    </div>
    <div
      style={Object {}}
    >
      <div
        className="bm-cross-button "
        style={
          Object {
            "height": "24px",
            "position": "absolute",
            "right": 8,
            "top": 8,
            "width": "24px",
          }
        }
      >
        <span
          style={
            Object {
              "position": "absolute",
              "right": "14px",
              "top": "6px",
            }
          }
        >
          <span
            className="bm-cross "
            style={
              Object {
                "background": "#bdc3c7",
                "height": 14,
                "position": "absolute",
                "transform": "rotate(45deg)",
                "width": 3,
              }
            }
          />
          <span
            className="bm-cross "
            style={
              Object {
                "background": "#bdc3c7",
                "height": 14,
                "position": "absolute",
                "transform": "rotate(-45deg)",
                "width": 3,
              }
            }
          />
        </span>
        <button
          onClick={[Function]}
          style={
            Object {
              "background": "transparent",
              "border": "none",
              "height": "100%",
              "left": 0,
              "margin": 0,
              "outline": "none",
              "padding": 0,
              "position": "absolute",
              "textIndent": -9999,
              "top": 0,
              "width": "100%",
            }
          }
        >
          Close Menu
        </button>
      </div>
    </div>
  </div>
  <div
    className="bm-burger-button "
    style={
      Object {
        "height": "60px",
        "left": "36px",
        "position": "absolute",
        "top": "36px",
        "width": "60px",
        "zIndex": 1,
      }
    }
  >
    <img
      className="bm-icon"
      src="./src/assets/menu.svg"
      style={
        Object {
          "height": "100%",
          "width": "100%",
        }
      }
    />
    <button
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      style={
        Object {
          "border": "none",
          "fontSize": 8,
          "height": "100%",
          "left": 0,
          "margin": 0,
          "opacity": 0,
          "padding": 0,
          "position": "absolute",
          "top": 0,
          "width": "100%",
        }
      }
    >
      Open Menu
    </button>
  </div>
</div>
`;
